pkg/sql/schemachanger/scpb/elements.proto

```diff
@@ -13,257 +13,225 @@ package cockroach.sql.schemachanger.scpb;
-  string name = 4;
-  string expr = 5 [(gogoproto.casttype)="github.com/cockroachdb/cockroach/pkg/sql/catalog/catpb.Expression"];
-  repeated uint32 column_ids = 6 [(gogoproto.customname) = "ColumnIDs", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.ColumnID"];
-  bool validated = 7;
+
+message Column {
+  uint32 table_id = 1 [(gogoproto.customname) = "TableID", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.DescID"];
+  uint32 family_id = 2 [(gogoproto.customname) = "FamilyID", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.FamilyID"];
+  uint32 column_id = 3 [(gogoproto.customname) = "ColumnID", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.ColumnID"];
+  TypeT embedded_type_t = 4 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  bool nullable = 6;
+  bool virtual = 7;
+  bool hidden = 8;
+  bool inaccessible = 9;
+  Expression compute_expr = 10;
+  uint32 generated_as_identity_type = 11 [(gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/catalog/catpb.GeneratedAsIdentityType"];
+  string generated_as_identity_sequence_option = 12;
+  uint32 pg_attribute_num = 13 [(gogoproto.customname) = "PgAttributeNum"];
+  uint32 system_column_kind = 14 [(gogoproto.customname) = "SystemColumnKind", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/catalog/catpb.SystemColumnKind"];
+}
+
+message ColumnFamily {
+  uint32 table_id = 1 [(gogoproto.customname) = "TableID", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.DescID"];
+  uint32 family_id = 2 [(gogoproto.customname) = "FamilyID", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.FamilyID"];
+  string name = 3;
 }
 
-message Sequence {
-  uint32 sequence_id = 1 [(gogoproto.customname) = "SequenceID", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.DescID"];
+message PrimaryIndex {
+  Index embedded_index = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
 }
 
-message DefaultExpression {
-  uint32 table_id = 1 [(gogoproto.customname) = "TableID", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.DescID"];
-  uint32 column_id = 2 [(gogoproto.customname) = "ColumnID", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.ColumnID"];
-  repeated uint32 uses_sequence_ids = 3 [(gogoproto.customname) = "UsesSequenceIDs", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.DescID"];
-  string default_expr = 4 [(gogoproto.casttype)="github.com/cockroachdb/cockroach/pkg/sql/catalog/catpb.Expression"];
+message SecondaryIndex {
+  Index embedded_index = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
 }
 
-message View {
-  uint32 table_id = 1 [(gogoproto.customname) = "TableID", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.DescID"];
+message SchemaParent {
```

Comment:
I'm adding a comment to explain these things in `elements.proto` but the short of it is, `SchemaParent` is responsible for updating the schema backrefs in the parent db throughout its lifecyle: remove the backref when targeting ABSENT and vice-versa.
Nothing's hard and fast here and this could conceivably be handled alternatively by `Namespace` or even `Schema` but I found this to be nicest.